Backend/problems/algebra.py

```diff
@@ -15,11 +18,44 @@ def modifyStep(step):
 
   return badStep
 
+def getProblem(activity):
+  #For now just get a problem out of the hardcoded 'problem bank'
+  problem = 'Bad Problem'
+  if activity in problemBank.keys():
+    problem = problemBank[activity]
+  return problem
+
+
+def getProblemTree(problem):
+  #When problem templates work problem could 
+  # be a problem template instead. If that's the case 
+  # we could get the problem from the template here
+  response = ControllerResponse()
+  numberOfFailurePoints = 2 #TODO: Replace this with the recommender system output
+
+  problemPath = mathsteps.getStepsForProblem(problem)
+  if len(problemPath) <= 1:
+    #We couldn't get a path for the problem
+    response.addError("Problem Solve Error", "Could not generate path for problem {0}".format(problem))
```

Comment:
Also if it returns no steps, that could just mean the problem is solved, not that there was an issue solving the problem.